homeassistant/components/sensor/mold_indicator.py

```diff
@@ -0,0 +1,255 @@
+        self._calc_dewpoint()
+        self._calc_moldindicator()
+        self.update_ha_state()
+
+    def _calc_dewpoint(self):
+        """Calculate the dewpoint for the indoor air."""
+        # use magnus approximation to calculate the dew point
+        alpha = MAGNUS_K2 * self._indoor_temp / (MAGNUS_K3 + self._indoor_temp)
+        beta = MAGNUS_K2 * MAGNUS_K3 / (MAGNUS_K3 + self._indoor_temp)
+
+        self._dewpoint = \
+            MAGNUS_K3 * (alpha + math.log(self._indoor_hum / 100.0)) / \
+            (beta - math.log(self._indoor_hum / 100.0))
+        _LOGGER.debug("Dewpoint: %f " + TEMP_CELCIUS, self._dewpoint)
+
+    def _calc_moldindicator(self):
+        """Calculate the humidity at the (cold) calibration point."""
+        if None in (self._dewpoint, self._calib_value) or \
+           self._calib_value == 0:
+
+            _LOGGER.debug("Invalid inputs - dewpoint: %s,"
+                          " calibration-Value: %s",
+                          self._dewpoint, self._calib_value)
+            self._state = "-"
+            return
+
+        # first calculate the approximate temperature at the calibration point
+        self._crit_temp = \
+            self._outdoor_temp + (self._indoor_temp - self._outdoor_temp) / \
+            self._calib_value
+
+        _LOGGER.debug(
+            "Estimated Critical Temperature: %f " +
+            TEMP_CELCIUS, self._crit_temp)
+
+        # Then calculate the humidity at this point
+        alpha = MAGNUS_K2 * self._crit_temp / (MAGNUS_K3 + self._crit_temp)
+        beta = MAGNUS_K2 * MAGNUS_K3 / (MAGNUS_K3 + self._crit_temp)
+
+        crit_humidity = \
+            math.exp(
+                (self._dewpoint * beta - MAGNUS_K3 * alpha) /
+                (self._dewpoint + MAGNUS_K3)) * 100.0
+
+        # check bounds and format
+        if crit_humidity > 100:
+            self._state = '100'
+        elif crit_humidity < 0:
+            self._state = '0'
+        else:
+            self._state = '{0:.2f}'.format(crit_humidity)
+
+        _LOGGER.debug('Mold indicator humidity: %s ', self._state)
+
+    @property
+    def should_poll(self):
+        """Polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return "%"
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        if self._dewpoint and self._crit_temp:
+
+            if self._is_metric:
+                return {
+                    ATTR_DEWPOINT: '{0:.2f}'.format(self._dewpoint) +
+                                   ' ' + TEMP_CELCIUS,
+                    ATTR_CRITICAL_TEMP: '{0:.2f}'.format(self._crit_temp) +
+                                        ' ' + TEMP_CELCIUS,
```

Comment:
Two decimal  places? For an estimation is an integer ok. The string concatenation looks a bit wired.